homeassistant/components/modbus/base_platform.py

```diff
@@ -130,13 +130,13 @@ def __init__(self, hub: ModbusHub, config: dict) -> None:
         super().__init__(hub, config)
         self._swap = config[CONF_SWAP]
         self._data_type = config[CONF_DATA_TYPE]
-        self._structure = config.get(CONF_STRUCTURE)
+        self._structure: str = config[CONF_STRUCTURE]
```

Comment:
Why this one is typed, but all the other assignments aren't?

Reply:
because it is not needed, for mypy-strict to work. mypy-strict does not say everything have to be typed.

Actually the ": str" is not needed, because like with the others it is inherited in the assigment.